MAINT validate parameter in Ridge

Eschivo opened the pull request:
#### Reference Issues/PRs
Towards #23462 


#### What does this implement/fix? Explain your changes.
Use of _validate_params for the Ridge family of algorithms

Eschivo committed:
Added _parameters_constraint to base Ridge

jeremiedbb commented on sklearn/linear_model/_ridge.py:
```diff
@@ -772,6 +775,22 @@ def _ridge_regression(
+    _parameter_constraints = {
+        "alpha": [Interval(Real, 1, None, closed="left"), float, np.ndarray],
+        "fit_intercept": [bool],
+        "normalize": [StrOptions({"deprecated"}), bool],
```
this should be `Hidden(StrOptions({"deprecated"}))` but needs this PR https://github.com/scikit-learn/scikit-learn/pull/23558 merged before

jeremiedbb commented on sklearn/linear_model/_ridge.py:
```diff
@@ -772,6 +775,22 @@ def _ridge_regression(
+        "fit_intercept": [bool],
+        "normalize": [StrOptions({"deprecated"}), bool],
+        "copy_X": [bool],
+        "max_iter": [Interval(Integral, 1, None, closed="left")],
```
it can also be None
```suggestion
        "max_iter": [Interval(Integral, 1, None, closed="left"), None],
```

jeremiedbb reviewed (commented):
Thanks for the PR @EliaSchiavon. Here are some suggestions.

The next step would to call `self._validate_params` in fit in `Ridge` and `RidgeClassifier` which inherit from this base class. Then check if previous validations are now obsolete and can be removed.

EliaSchiavon commented:
Thank you very much for the feedback, Jeremie. I'm gonna implement all the things the next week since I'm away during the weekend.

Eschivo committed:
Parameter constraints in _BaseRidge class completed

Eschivo committed:
self.validate_params in Ridge and RidgeClssifier

Eschivo committed:
Removed obsolete params testing from fit in _BaseRidge

Eschivo committed:
Extende parameter contraints in RidgeClassifier + minor fix in parameter constraints _BaseRidge

Eschivo committed:
Update test_common.py

-removed Ridge and RidgeClassifier from list of PARAM_VALIDATION_ESTIMATORS_TO_IGNORE

Eschivo committed:
Minor update to ridge.py

normalize in parameter constraints in BaseRidge is now an Hidden
StrOptions

Eschivo committed:
Minor fix in _ridge.py

EliaSchiavon commented:
All the things should be ok for Ridge and RidgeClassifier. The only thing failing in the test is the implementation of the normalize parameter as Hidden, as you suggested in the comments above.

jeremiedbb committed:
update

jeremiedbb committed:
Merge remote-tracking branch 'upstream/main' into pr/EliaSchiavon/23563

jeremiedbb committed:
remove and fix test

jeremiedbb reviewed (approved):
LGTM

glemaitre commented on sklearn/linear_model/_ridge.py:
```diff
@@ -637,7 +641,7 @@ def _ridge_regression(
     # Some callers of this method might pass alpha as single
     # element array which already has been validated.
-    if alpha is not None and not isinstance(alpha, (np.ndarray, tuple)):
+    if alpha is not None and not isinstance(alpha, (np.ndarray)):
```
we can probably remove the paranthesis around `np.ndarray`.
We are sure that we don't have any case where the value in the `tuple` have been validated but not transformed to a NumPy array.

glemaitre reviewed (commented):
Otherwise LGTM.

jeremiedbb committed:
Update sklearn/linear_model/_ridge.py

EliaSchiavon commented:
Thank you for your work @jeremiedbb and @glemaitre !
